src/librustdoc/test.rs

```diff
@@ -293,6 +293,39 @@ fn run_test(
             path.to_str().expect("target path must be valid unicode").to_string()
         }
     });
+    match options.error_format {
+        ErrorOutputType::HumanReadable(kind) => {
+            let (_, color_config) = kind.unzip();
+            match color_config {
+                ColorConfig::Never => {
+                    compiler.arg("--color").arg("never");
+                }
+                ColorConfig::Always => {
+                    compiler.arg("--color").arg("always");
+                }
+                ColorConfig::Auto => {
+                    #[cfg(windows)]
+                    {
+                        // This specific check is because old windows consoles require a connection
+                        // to be able to display colors (and they don't support ANSI), which we
+                        // cannot in here, so in case this is an old windows console, we can't
+                        // display colors.
+                        use crate::termcolor::{ColorChoice, StandardStream, WriteColor};
+                        if StandardStream::stdout(ColorChoice::Auto).is_synchronous() {
+                            compiler.arg("--color").arg("never");
+                        } else {
+                            compiler.arg("--color").arg("always");
+                        }
+                    }
+                    #[cfg(not(windows))]
+                    {
+                        compiler.arg("--color").arg("always");
+                    }
```

Comment:
I think this is missing the check that this is a TTY in both the windows and non-windows case. That's why I was suggesting to query the Emitter which has already figured these things out (and to avoid duplicating that logic).